### ISSUE TITLE
Type inferred as `any` in `.svelte` file.

### ISSUE THREAD

**Issue by Stadly** [sveltejs/eslint-plugin-svelte#713]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.57.0

### What version of `eslint-plugin-svelte` are you using?

2.35.1

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = {
  extends: ['plugin:@typescript-eslint/recommended-type-checked'],
  overrides: [
    {
      files: ['*.svelte'],
      parser: 'svelte-eslint-parser',
      parserOptions: {
        parser: '@typescript-eslint/parser',
        project: true,
        tsconfigRootDir: __dirname,
      },
    },
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    sourceType: 'module',
    extraFileExtensions: ['.svelte'],
    project: ['tsconfig.json', 'tsconfig.node.json'],
    tsconfigRootDir: __dirname,
  },
};

```
</details>

```svelte
<script lang="ts">
  export let transition:
    | {
        duration?: number;
      }
    | undefined = undefined;
  if (transition) {
    // transition.duration should not be inferred as any but number
    transition.duration = transition.duration ?? 1500;
  }
</script>

```

### What did you expect to happen?

`transition.duration` should be inferred as `number`.

If you look at `transition.ts` in the reproduction, you'll see it works correctly for `.ts` files, so it seems to be an issue with `.svelte` files.

### What actually happened?

`transition.duration` is inferred as `any`.

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/Stadly/reproduction/tree/eslint-plugin-svelte/type-inferred-as-any
